### PR TITLE
add Logs URL to html test report and email report

### PR DESCRIPTION
**Pull request by dahorak**
- if RUN["logs_url"] is defined, include it to the Environment part of html
  and email report and fix Log File links to point ot the logs_url
  instead of local log_dir
- fixes: https://github.com/red-hat-storage/ocs-ci/issues/3511

Signed-off-by: Daniel Horak <dahorak@redhat.com>

### CODE REVIEW

File: ocs_ci/framework/pytest_customization/ocscilib.py [hunk 1]
```diff
@@ -264,6 +264,11 @@ def pytest_configure(config):
         # Add OCS related versions to the html report and remove
         # extraneous metadata
         markers_arg = config.getoption("-m")
+
+        # add logs url
+        if ocsci_config.RUN.get("logs_url"):
```
Comment (vavuthu) [inline ocs_ci/framework/pytest_customization/ocscilib.py:269]: are we forming logs_url somewhere in code or passing through yaml?

File: ocs_ci/framework/pytest_customization/ocscilib.py [hunk 2]
```diff
@@ -264,6 +264,11 @@ def pytest_configure(config):
...
+            config._metadata["Logs URL"] = ocsci_config.RUN.get("logs_url")
```
Comment (petr-balogh) [inline ocs_ci/framework/pytest_customization/ocscilib.py:270]:
```suggestion
        logs_url = ocsci_config.RUN.get("logs_url")
        if logs_url:
            config._metadata["Logs URL"] = logs_url
```

File: ocs_ci/framework/pytest_customization/ocscilib.py (hunk 1 above)
Comment (petr-balogh) [reply ocs_ci/framework/pytest_customization/ocscilib.py:269]: Please document this value in:  https://github.com/red-hat-storage/ocs-ci/blob/master/conf/README.md

Also other request which @mbukatov will appreciate is to add this to xunit report. Can be done by other PR but please open issue to do that later on if not done in this PR.

Thanks!

File: ocs_ci/framework/pytest_customization/ocscilib.py (hunk 2 above)
Comment (dahorak) [reply ocs_ci/framework/pytest_customization/ocscilib.py]: Fixed by 6ab591a.

File: ocs_ci/framework/pytest_customization/ocscilib.py (hunk 1 above)
Comment (dahorak) [reply ocs_ci/framework/pytest_customization/ocscilib.py]: `logs_url` will is configured by Jenkins job (see [MR!468](https://gitlab.cee.redhat.com/ocs/ocs4-jenkins/-/merge_requests/468))

**Review by vavuthu — DISMISSED** [review]

File: ocs_ci/framework/pytest_customization/reports.py [hunk 3]
```diff
@@ -23,6 +23,17 @@ def pytest_html_results_table_row(report, cells):
...
+    if ocsci_config.RUN.get("logs_url"):
+        for tag in cells[4][0]:
+            if (
+                hasattr(tag, "xmlname")
+                and tag.xmlname == "a"
+                and hasattr(tag.attr, "href")
+            ):
+                tag.attr.href = tag.attr.href.replace(
+                    ocsci_config.RUN.get("log_dir"), ocsci_config.RUN.get("logs_url")
+                )
```
Comment (mbukatov) [inline ocs_ci/framework/pytest_customization/reports.py:36]: Ah, so this fixes old known issue OCSQE-350.

File: conf/README.md [hunk 4]
```diff
@@ -41,6 +41,7 @@ run it belongs here.
 * `username` - Kubeadmin username
 * `password_location` - Filepath (under the cluster path) where the kubeadmin password is located
 * `log_dir` - Directory where logs are placed
+* `logs_url` - URL where will be available logs for remote access, used for Jenkins runs
```
Comment (mbukatov) [inline conf/README.md:44]: Could you note there that this value is generated and passed by jenkins?

**Review by mbukatov — APPROVED** [review]

**Comment by dahorak** [comment]
I have to fix small issue with not expanding `~` to home directory in the `log_dir`.

**Review by mbukatov — DISMISSED** [review]

File: conf/README.md [hunk 5]
```diff
@@ -41,6 +41,7 @@ run it belongs here.
 * `username` - Kubeadmin username
 * `password_location` - Filepath (under the cluster path) where the kubeadmin password is located
 * `log_dir` - Directory where logs are placed
+* `logs_url` - URL where will be available logs for remote access, used for Jenkins runs and configured by Jenkins
```
Comment (vasukulkarni) [inline conf/README.md:44]: rewrite - URL where the logs will be available for remote access,

Comment (dahorak) [reply conf/README.md]: Updated, I was thinking about the word _rewrite_, but it is used also in the *Environment* section, where it doesn't rewriting anything in fact. So I change the wording, but without the _rewrite_.

Comment (vasukulkarni) [reply conf/README.md]: i didn't mean to include word rewrite but re-word the description :)

**Review by vasukulkarni — APPROVED** [review]

**Comment by dahorak** [comment]
_Logs URL_ is present in the *Environment* table:
pr3512a_verification
and _Log File_ link points to correct url:
pr3512b_verification

**Review by mbukatov — APPROVED** [review]

**Review by clacroix12 — APPROVED** [review]

**Pull request merged**